internal/graph/graph.go

```diff
@@ -52,120 +50,101 @@ type ResolutionMetadata struct {
 	} else {
-		val = fmt.Sprintf("ingress %s, type %s", r.Ingress.String(), r.Type.String())
+		val = fmt.Sprintf("userset %s, type %s", r.TailUsersetNode.String(), r.Type.String())
 	}
 	return strings.ReplaceAll(val, "  ", " ")
 }
 
-// ConnectedObjectGraph represents a graph of relationships and the connectivity between
+// TypesystemGraph represents a graph of relationships and the connectivity between
 // object and relation references within the graph through direct or indirect relationships.
-// The ConnectedObjectGraph should be used to introspect what kind of relationships between
-// object types can exist.
-type ConnectedObjectGraph struct {
+type TypesystemGraph struct {
 	typesystem *typesystem.TypeSystem
 }
 
-// BuildConnectedObjectGraph builds an object graph representing the graph of relationships between connected
-// object types either through direct or indirect relationships.
-func BuildConnectedObjectGraph(typesystem *typesystem.TypeSystem) *ConnectedObjectGraph {
-	return &ConnectedObjectGraph{
+// New returns a TypesystemGraph. The TypesystemGraph should be used to introspect what kind of relationships between
+// object types can exist. To visualize this graph, use https://github.com/jon-whit/openfga-graphviz-gen
+func New(typesystem *typesystem.TypeSystem) *TypesystemGraph {
+	return &TypesystemGraph{
 		typesystem: typesystem,
 	}
 }
 
-// RelationshipIngresses computes the incoming edges (ingresses) that are possible between the target relation reference
-// and the source relational reference.
-//
-// To look up Ingresses(`document#viewer`, `source`), where `source` is a type with no relation, look up the definition of relation `document#viewer` and:
-// 1. If `source` is a directly related type then add `source, direct` to the result.
-// 2. If `objectType#relation` is a type and `objectType#relation` can be a `source` then add `objectType#relation, direct` to the result.
-// 3. If computed userset, say `define viewer as writer`, then recurse on `document#writer, source`.
-// 4. If tuple-to-userset, say, `define viewer as viewer from parent`. Go to parent and find its types. In this case, `folder`. Go to `folder` and see if it has a `viewer` relation. If so, recurse on `folder#viewer, source`.
-//
-// To look up Ingresses(`document#viewer`, `folder#viewer`), look up the definition of relation `document#viewer` and:
-// 1. If `folder#viewer` is a directly related type then add `folder#viewer, direct` to the result.
-// 2. If computed userset, say `define viewer as writer`, then recurse on `document#writer, folder#viewer`.
-// 3. If tuple-to-userset, say, `define viewer as viewer from parent`. Go to parent and find its related types.
-//  1. If parent's types includes `folder` type, and `folder` contains `viewer` relation then this is exactly a ttu rewrite and....?
-//  2. Otherwise, suppose the types contains `objectType` which has a relation `viewer`, then recurse on `objectType#viewer, folder#viewer`
```

Comment:
I feel it's 100% easier to understand by looking at a graph than by reading this chunk